node/tests/RedisClientInternals.test.ts

```diff
@@ -627,4 +627,147 @@ describe("SocketConnectionInternals", () => {
             expect(result2).toBeNull();
         });
     });
+
+    it("should set arguments according to xadd options", async () => {
+        let counter = 0;
+        await testWithClusterResources(async (connection, socket) => {
```

Comment:
Please split into 4 different test cases

Reply:
once we'll have the utilities not to have a lot of wrapping around the assert, sure. not now.